### PR TITLE
Fix some issues in Swap mode and update to latest swap features

### PR DESCRIPTION
**Pull request by fbeutin-ledger**
Fix BSS handling in SWAP mode
Split swap features in several files
Reuse SWAP interface from SDK
Add a modal on Stax startup in SWAP mode
Return signing status to Exchange after a SWAP completion

:link: [zboto Link](https://app.clickup.com/t/8692tuf92)

### CODE REVIEW

**Comment by fbeutin-ledger** [comment]
Needs https://github.com/Zondax/ledger-zxlib/pull/113 first

File: app/src/swap/handle_check_address.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        MEMZERO(buffer, sizeof(buffer));
+        return;
+    }
+
+    const uint8_t *address = buffer + PK_LEN_25519;
+    const uint8_t addressLen = replyLen - PK_LEN_25519;
+    const uint8_t addressToCheckLen = strlen(params->address_to_check);
+
+    if (addressLen == addressToCheckLen &&
+        memcmp(address, params->address_to_check, addressLen) == 0) {
```
Comment (ftheirs) [inline app/src/swap/handle_check_address.c:52]: using `strnlen` would be better here

File: app/src/common/main.c [hunk 2]
```diff
@@ -31,38 +37,84 @@ static void app_exit(void) {
...
+                    // app-exchange BSS data.
+                    bool success = copy_transaction_parameters(args->create_transaction);
+                    if (success) {
+                        // BSS was wiped, we can now init these globals
+                        G_swap_state.called_from_swap = true;
+                        G_swap_state.should_exit = false;
+
+#ifdef HAVE_NBGL
+                        // On Stax, display a modal
+                        nbgl_useCaseSpinner("Signing");
```
Comment (ftheirs) [inline app/src/common/main.c:56]: Internal note: we can add in zxlib a spinner/loading interface to avoid using UI methods directly

**Comment by ftheirs** [comment]
Hi @fbeutin-ledger! I fixed some minor stuff and the CI is almost fully in green. I had to open this PR to NanoS SDK
https://github.com/LedgerHQ/nanos-secure-sdk/pull/63
Can you take a look there? Once we have that fix, the PR should be merged 💪

File: app/src/swap/handle_check_address.c (hunk 1 above)
Comment (fbeutin-ledger) [reply app/src/swap/handle_check_address.c:52]: Done

**Pull request closed without merging**